server.js

```diff
@@ -16,9 +20,66 @@ const app = express();
 app.use(cors());
 app.use(express.json());
 
+// Mongoose model
+const Thought = mongoose.model("Thought", {
+  message: {
+    type: String,
+    required: true,
+    maxlength: 140,
+    minlength: 5,
+  },
+  hearts: { type: Number, default: 0 },
+  createdAt: { type: Date, default: Date.now },
+});
+
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  const endpoints = expressListEndpoints(app);
+  res.json(endpoints);
+});
+
+app.get("/thoughts", async (req, res) => {
+  // Get all the thoughts
+  const thoughts = await Thought.find().sort(
+    { createdAt: "desc" }).limit(20).exec();
+
+  try {
+    res.status(201).json(thoughts);
+  } catch (err) {
+    res.status(400).json({
+      message: "Could not retrieve the Happy Thoughts.",
```

Comment:
Interesting, we used different status codes here 😄

In "try" I used 200 as I read it's a more general "OK" status for GET requests while 201 specifically means "created" when you have done a successful POST request. 

In "catch" I used 500 which indicates server-side errors (such as database query error, ,which would likely be the case here). 

But again, might be wrong here too! By no means I feel comfortable with these status codes, just find it interesting that we chose different ones :)